app/bundles/CampaignBundle/Entity/CampaignRepository.php

```diff
@@ -426,15 +430,21 @@ public function getPendingContactIds($campaignId, ContactLimiter $limiter)
         )
             ->setParameter('campaignId', (int) $campaignId);
 
-        $results = $q->execute()->fetchAll();
+        if ($limiter->hasCampaignLimit()) {
+            $q->setMaxResults($limiter->getCampaignLimitRemaining());
```

Comment:
Doesn't the query already have a max limit from the batch count? If a max limit already exists we'd want to decrease it only if our limit is lower. This should be in updateQueryFromContactLimiter not here.

Reply:
Good call. Hadn't considered that

Reply:
Using values from the limiter outside the limiter to modify queries is contrary to the pattern established.
Please move this logic into updateQueryFromContactLimiter.

Reply:
I cannot. It is only at the repository level I know if campaign is being considered. Within the limiter, it is only considering the contacts.
Also ContactLimiterTrait uses an instance of ContactLimiter to do the same thing...
`
        if (!$isCount && $limit = $contactLimiter->getBatchLimit()) {
            $qb->setMaxResults($limit);
        }
`